packages/utils/src/cmds/schema/diff/index.js

```diff
@@ -0,0 +1,314 @@
+const path = require('path')
+const jsondiffpatch = require('jsondiffpatch')
+const fs = require('fs')
+const utils = require('./utils')
+const ejs = require('ejs')
+const request = require('request')
+const { reporter } = require('@dhis2/cli-helpers-engine')
+const inquirer = require('inquirer')
+
+const defaultOpts = {
+    schemasEndpoint: '/api/schemas.json',
+    infoEndpoint: '/api/system/info.json',
+}
+
+const defaultRequestOpts = {
+    baseUrl: 'https://play.dhis2.org',
+    headers: {
+        'x-requested-with': 'XMLHttpRequest',
+        Authorization: utils.basicAuthHeader('admin', 'district'),
+    },
+    json: true,
+}
+
+let cache
+const prompt = inquirer.createPromptModule({ output: process.stderr })
+
+const schemaIdentifier = info => `${info.version}_${info.revision}`
+const schemaDiffIdentifier = (info1, info2) =>
+    `${schemaIdentifier(info1)}__${schemaIdentifier(info2)}`
+
+// We use the singular property as an unique identifier for schemas
+// name and type are used for other nested properties, fallback to index
+const Differ = jsondiffpatch.create({
+    objectHash: obj =>
+        obj.singular || obj.name || obj.type || '$$index:' + index,
+    propertyFilter: name => name !== 'href' && name !== 'apiEndpoint',
+    arrays: {
+        detectMove: true,
+        includeValueOnMove: true,
+    },
+})
+
+const formatters = {
+    html: jsondiffpatch.formatters.html,
+    console: jsondiffpatch.formatters.console,
+}
+
+function asyncRequest(url, opts) {
+    return new Promise((resolve, reject) => {
+        request.get(url, opts, (err, res, body) => {
+            if (err || res.statusCode > 299) {
+                reporter.error('Request', url, 'failed to fetch')
+                err && reporter.dumpErr(err)
+                res &&
+                    reporter.error(
+                        res.statusCode,
+                        ':',
+                        res.statusMessage || (res.body && res.body.message)
+                    )
+                process.exit(1)
+            }
+            return resolve(body)
+        })
+    })
+}
+
+function getJSONFile(file) {
+    try {
+        const content = fs.readFileSync(file)
+        return JSON.parse(content)
+    } catch (e) {
+        return false
+    }
+}
+
+async function getAuthHeader(url, { auth }) {
+    let username, password
+    if (auth) {
+        ;[username, password] = auth.split(':')
+    }
+    if (auth === '') {
+        ;({ username, password } = await prompt([
+            {
+                type: 'input',
+                name: 'username',
+                message: `Username for ${url}`,
+                default: 'admin',
```

Comment:
This is fine for now but at some point we'll probably want to "centralize" the defaults for cli tools (where to find the server you want to talk to, what auth to use), so we can easily change/remove them or at least make them configurable.